src/lib/uri/uri.c

```diff
@@ -72,3 +181,21 @@ uri_format(char *str, int len, const struct uri *uri, bool write_password)
 	}
 	return total;
 }
+
+const char *
+uri_query_param(const struct uri *uri, const char *name, int idx)
+{
+	struct uri_query_param *param =
+		uri_find_query_param(uri, name);
+	if (param == NULL || idx < 0 || idx >= param->value_count)
```

Comment:
Nit: `assert(idx >= 0)` would be more appropriate, as a negative index doesn't make any sense.

Reply:
fixed